UE4SS/src/UE4SSProgram.cpp

```diff
@@ -1117,67 +1117,127 @@ namespace RC
+            // Parse the line into something that can be converted into proper data
+            std::wstring mod_name = explode_by_occurrence(current_line, L':', 1);
+            std::wstring mod_enabled = explode_by_occurrence(current_line, L':', ExplodeType::FromEnd);
+            bool mod_enabled_out = !mod_enabled.empty() && mod_enabled[0] == L'1';
 
-                auto mod = UE4SSProgram::find_mod_by_name<ModType>(mod_name, UE4SSProgram::IsInstalled::Yes);
-                if (!mod || !dynamic_cast<ModType*>(mod))
-                {
-                    continue;
-                }
+            auto it = std::find_if(mod_data_vector.begin(), mod_data_vector.end(),
+                           [&mod_name](const ModData& element) {
+                               return element.mod_name == to_string(mod_name);
+                           });
 
-                if (!mod_enabled.empty() && mod_enabled[0] == L'1')
-                {
-                    Output::send(STR("Starting {} mod '{}'\n"), std::is_same_v<ModType, LuaMod> ? STR("Lua") : STR("C++"), mod->get_name().data());
-                    mod->start_mod();
-                }
-                else
-                {
-                    Output::send(STR("Mod '{}' disabled in mods.txt.\n"), mod_name);
-                }
+            if (it == mod_data_vector.end()) {
+                mod_data_vector.push_back({to_string(mod_name), mod_enabled_out});
             }
         }
+        
+        Output::send(STR("mods.txt entries converted to mods.json.  Please delete mods.txt if no mod managers rely on it to avoid conflicting with changes to the json...\n"));
+    }
+
 
-        // Part #2: Start all mods that have enabled.txt present in the mod directory.
-        Output::send(STR("Starting mods (from enabled.txt, no defined load order)...\n"));
+    template <typename ModType>
+    auto start_mods() -> std::string
+    {
+        ProfilerScope();
+        // Part #1: Start all mods that are enabled in mods.json.
+        std::filesystem::path mods_directory = UE4SSProgram::get_program().get_mods_directory();
+        std::filesystem::path legacy_mod_list_file{mods_directory / "mods.txt"};
+        std::filesystem::path mod_list_file{mods_directory / "mods.json"};
+        std::vector<UE4SSProgram::ModData> mod_data_vector{};
+        UE4SSProgram& program = UE4SSProgram::get_program();
+
+        if (std::filesystem::exists(mod_list_file))
+        {
+            UE4SSProgram::read_mods_json(mod_list_file.string(), mod_data_vector);
+        }
+        
+        if (std::filesystem::exists(legacy_mod_list_file))
+        {
+            UE4SSProgram::convert_legacy_mods_file(legacy_mod_list_file, mod_data_vector);
+        }
+        
+        UE4SSProgram::write_mods_json(mod_list_file.string(), mod_data_vector);
+
+        
+        Output::send(STR("Starting mods (from mods.json load order)...\n"));
+        for (auto it = mod_data_vector.begin(); it != mod_data_vector.end(); ++it)
```

Comment:
Any reason why we're using the more complicated iterator method of looping here instead of the simpler
`for (const auto& mod_data : mod_data_vector)` ?